example-tests/src/test/java/com/epam/restassured/csvreader/CSVReaderUtilitySingleton.java

```diff
@@ -0,0 +1,105 @@
+package com.epam.restassured.csvreader;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.epam.restassured.pojo.csv.CSVTestInput;
+import com.opencsv.CSVReader;
+
+/**
+ * Singleton representation of a CSV Reader utility class.
+ * 
+ * Please note, CSV input files should follow the following format:
+ * No comment line, data should start in the first line in the following oder
+ * FirstName,LastName,EmailAddress,EmailAddressConfirmation,NewsletterOptIn
+ * 
+ * Open issue for OpenCSV to add comment line in the first line of the input CSV file
+ * 
+ * @author Tamas_Csako
+ *
+ */
+public class CSVReaderUtilitySingleton {
+	
+	//Default file name to read input data
+	private static final String DEFAULT_TEST_INPUT_FILE = "test_data_rest.csv";
+	//Private instance
+	private static CSVReaderUtilitySingleton instance = null;
+	
+	//Protected constructor for the Singleton implementation
+	protected CSVReaderUtilitySingleton(){
+	}
+	
+	/**
+	 * Public method to get instance for the Singleton class
+	 * 
+	 * @return instance of the Singleton object
+	 */
+	public static CSVReaderUtilitySingleton getInstance() {
+
+		if (instance == null) {
+			instance = new CSVReaderUtilitySingleton();
+		}
+		
+		return instance;
+	}
+
+	/**
+	 * Method to get test input data from resource file.
+	 * 
+	 * @param fileName - file name and path to read input data
+	 * @return list of input data in CSVTestInput format
+	 */
+	@SuppressWarnings("resource")
+	public List<CSVTestInput> getIntput(String fileName) {
```

Comment:
This is too specific. It should be either generic or return a list of Strings for now.